zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -156,10 +158,18 @@
      */
     private final Set<String> containers = Collections.newSetFromMap(new ConcurrentHashMap<String, Boolean>());
 
+    private final Map<String, TTLNode> ttlMap = new HashMap<>();
     /**
-     * This set contains the paths of all ttl nodes
+     * This set contains the paths of all ttl nodes, ranking by expireTime asc.
      */
-    private final Set<String> ttls = Collections.newSetFromMap(new ConcurrentHashMap<String, Boolean>());
+    public final Set<TTLNode> ttls = new TreeSet<>(
```

Comment:
Did you consider using a `PriorityQueue` instead of a TreeSet? It seems like a more appropriate data structure.